x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/TextExpansionQueryIT.java

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.integration;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.xpack.core.ml.utils.MapHelper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * This test uses a tiny text embedding model to simulate a trained
+ * NLP model. The output tensor is randomly generated but the RNG is
+ * seeded with the hash of the input IDS so the same input produces
+ * the same output.
+ *
+ * The model was created using this simple Python code.
+ * The saved TorchScript model is then base64 encoded and hardcoded here
+ * for use in the test.
+ *
+ * ## Start Python
+ * import torch
+ * from torch import Tensor
+ * from typing import Optional
+ *
+ * class TinyTextEmbedding(torch.nn.Module):
+ *
+ *  def forward(self,
+ *          input_ids: Tensor,
+ *          token_type_ids: Optional[Tensor] = None,
+ *          position_ids: Optional[Tensor] = None,
+ *          inputs_embeds: Optional[Tensor] = None):
+ *
+ *      torch.random.manual_seed(hash(str(input_ids)))
+ *      return torch.rand(1, 100)
+ *
+ * if __name__ == '__main__':
+ *      tte = TinyTextEmbedding()
+ *      tte.eval()
+ *      input_ids = torch.tensor([1, 2, 3, 4, 5])
+ *      the_rest = torch.ones(5)
+ *      traced_model =  torch.jit.script(tte, (input_ids, the_rest, the_rest, the_rest))
+ *      torch.jit.save(traced_model, "simplemodel.pt")
+ * ## End Python
+ */
+public class TextExpansionQueryIT extends PyTorchModelRestTestCase {
```

Comment:
Integration test name seems off...but that is a minor thing.

Reply:
Yes I er did this on purpose to make a point about how easy it is to confuse embedding and expansion. Well spotted.